### DISCUSSION TITLE
Problem of using MatchedValueBC

### DISCUSSION THREAD

**Discussion by SomnusYu** [idaholab/moose#27552] [Q&A General]
Hi all,

I would like to know if each variable must have one kernel at least. Can I just use the `MatchedValueBC` of this variable, but do not use kernel of this variable (that's to say the value of this variable equal to `v ` in `MatchedValueBC` in each linear and nonlinear iteration)? 

Thanks!

**Comment by GiudGiud** [comment]
Hello

Yes that's pretty much an obligation because having no kernels means huge rows of 0s in the Jacobian.
Just add a time derivative kernel on that variable, that does not change much

Guillaume

**Reply by SomnusYu** [reply]
I have tried to add a `ADTimeDerivative` kernel, but the value equals to initial value and doesn't change. Is there any other kernel I can use?

**Reply by GiudGiud** [reply]
why would it change? From the BC? The volume would stay constant for sure with just this kernel

**Reply by SomnusYu** [reply]
Yes, the `v` in `MatchedValueBC` will change over the time.

**Reply by GiudGiud** [reply]
and you want u to be equal to v everywhere in the domain?
or just on the boundary?

**Reply by SomnusYu** [reply]
Just on the boundary.

**Reply by GiudGiud** [reply]
and you are not seeing that happen? as if the BC is not being applied?

you could try this boundary condition too. v is a Functor (since it s a variable)
https://mooseframework.inl.gov/source/bcs/FunctorDirichletBC.html

**Reply by SomnusYu** [reply]
OK, thanks a lot!
